src/test/java/software/amazon/encryption/s3/internal/ContentMetadataStrategyTest.java

```diff
@@ -52,7 +52,7 @@ public void decodeWithObjectMetadata() {
                 .contentIv(bytes)
                 .build();
 
-        ContentMetadata contentMetadata = ContentMetadataStrategy.decode(getObjectRequest, getObjectResponse);
+        ContentMetadata contentMetadata = new ContentMetadataDecodingStrategy(null).decode(getObjectRequest, getObjectResponse);
```

Comment:
(nit) Seems this should be a mock S3 client instead of null; if this test were to attempt to call S3 it would result in NPE.
But this test doesn't call S3, so this is a nit.

Reply:
Do we need null checking somewhere?
If I pass in a null S3 client due to being (silly, adversarial, etc.), is this checked/guarded against in the non-internal class level?

Reply:
The top-level client builders will create a client when there is none provided, 
so it is always the case that the client is not null. 
That said,
I'll add a null check / mock in the name of robustness.

Reply:
Cool -- I think creating a default top-level client is sufficient but more checks never hurt anyone 👍